File: VNC.CodeAnalysis Test Cases/TestCasesCS/Quality/ControlFlags.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSTestCases.Quality
{
    class ControlFlags
    {
        bool cflag = false;

        public void update()
        {
            if (!cflag)
            {
                if (thatThing())
                    cflag = true;
            }
            else
            {
                thatOtherThing();
            }
        }

        void thatOtherThing()
        {
            bool flag = false;
            if (flag)
            {
                //Do that
            }
            else
{
                //Do something else
                flag = false;
            }
        }

        Boolean thatThing()
        {
            return true;
        }


    }
}

File: VNC.CodeAnalysis Test Cases/TestCasesVB/Quality/ControlFlags.vb
Public Class ControlFlags
    Private cflag As Boolean = False

    Public Sub update()
        Dim flag As Boolean = Not Me.cflag
        If flag Then
            Dim flag2 As Boolean = Me.thatThing()
            If flag2 Then
                Me.cflag = True
            End If
        Else
            Me.thatOtherThing()
        End If
    End Sub

    Private Sub thatOtherThing()
        Dim flag As Boolean = False
        Dim flag2 As Boolean = flag
        If flag2 Then
        End If
    End Sub

    Private Function thatThing() As Boolean
        Return True
    End Function

End Class
